cfgov/v1/models/base.py

```diff
@@ -203,8 +206,16 @@ def get_appropriate_siblings(self, inclusive=True):
 
     def get_context(self, request, *args, **kwargs):
         context = super(CFGOVPage, self).get_context(request, *args, **kwargs)
+
         for hook in hooks.get_hooks('cfgovpage_context_handlers'):
             hook(self, request, context, *args, **kwargs)
+
+        context['banners'] = []
+        enabled_banners = Banner.objects.filter(enabled=True)
```

Comment:
It's possible to simplify and optimize this further to do the regex matching as part of the database query:

```py
# Add any banners that are enabled and match the current request path.
context['banners'] = Banner.objects \
    .filter(enabled=True) \
    .annotate(path=Value(
        request.path,
        output_field=models.CharField()
    )) \
    .filter(path__regex=F('url_patterns'))
```

This creates an SQL query like this:

```sql
SELECT
    "v1_banner"."id",
    "v1_banner"."title",
    "v1_banner"."url_patterns",
    "v1_banner"."content",
    "v1_banner"."enabled",
    /about-us/careers/ AS "path"
FROM
    "v1_banner"
WHERE (
    "v1_banner"."enabled" = True AND
    /about-us/careers/::text ~ ("v1_banner"."url_patterns")
)
```

Here a banner has been defined with a regex `/about-us/`, and I'm visiting the page http://localhost:8000/about-us/careers/.

Reply:
This seems like a great optimization! I will incorporate. Thanks!